src/core/ext/filters/client_channel/resolver/dns/c_ares/dns_resolver_ares.c

```diff
@@ -152,8 +233,39 @@ static void dns_ares_on_resolved_locked(grpc_exec_ctx *exec_ctx, void *arg,
   r->resolving = false;
   r->pending_request = NULL;
   if (r->lb_addresses != NULL) {
-    grpc_arg new_arg = grpc_lb_addresses_create_channel_arg(r->lb_addresses);
-    result = grpc_channel_args_copy_and_add(r->channel_args, &new_arg, 1);
+    static const char* args_to_remove[2];
+    size_t num_args_to_remove = 0;
+    grpc_arg new_args[3];
+    size_t num_args_to_add = 0;
+    new_args[num_args_to_add++] =
+        grpc_lb_addresses_create_channel_arg(r->lb_addresses);
+    grpc_service_config *service_config = NULL;
+    if (r->service_config_json != NULL) {
+      char* service_config_string =
```

Comment:
We may need to free `service_config_string` after `grpc_channel_args_copy_and_add_and_remove()` is called.